autocert/controller/main.go

```diff
@@ -385,10 +402,26 @@ func shouldMutate(metadata *metav1.ObjectMeta) bool {
 	// Only mutate if the object is annotated appropriately (annotation key set) and we haven't
 	// mutated already (status key isn't set).
 	if annotations[admissionWebhookAnnotationKey] == "" || annotations[admissionWebhookStatusKey] == "injected" {
-		return false
+		return false, nil
+	}
+
+	if !restrictToNamespace {
+		return true, nil
+	}
+
+	subject := strings.Trim(annotations[admissionWebhookAnnotationKey], ".")
+
+	err := fmt.Errorf("subject \"%s\" matches a namespace other than \"%s\" and is not permitted. This check can be disabled by setting restrictCertificatesToNamespace to false in the autocert-config ConfigMap", subject, namespace)
+
+	if strings.HasSuffix(subject, ".svc") && !strings.HasSuffix(subject, fmt.Sprintf(".%s.svc", namespace)) {
```

Comment:
@justinbarrick I'm not an expert on this @mmalone should now better, but is it common to have a DNS entry that finishes in `.namespace.svc`? My understanding was they finish in `.namespace.svc.clusterDoman`

If this is the case, you will never get to the last `return true, nil`

Reply:
In kubernetes the search domain gets set to:

```
namespace.svc.cluster.local svc.cluster.local cluster.local
```

This means that for service `hello` in namespace `world` all of the following are valid, resolvable hostnames:

* If you are in the `world` namespace, `hello`.
* `hello.world`
* `hello.world.svc`
* `hello.world.svc.cluster.local`

I’m not sure what your last sentence means, but I don’t think that is accurate and the test cases confirm the logic.

Reply:
Got it didn't read the condition properly, I skipped the first part. I will wait for @mmalone to take a look at this before merging. Thanks